File: lista1/LiczbyPierwszeTest.java
public class LiczbyPierwszeTest {
    public static void main( String args[]) {
        try {
            int x = Integer.parseInt(args[0]);
            LiczbyPierwsze tester = new LiczbyPierwsze(x);
            for (int i = 1; i<args.length; i++){
                try {
                    int n = Integer.parseInt(args[i]);
                    if (tester.liczba(n)!= -1){
                        System.out.println(n + " - "+ tester.liczba(n));
                    }
                    else {
                        System.out.println(n + " - liczba spoza zakresu");
                    }
                }
                catch(NumberFormatException ex) {
                    System.out.println(args[i] + " - nieprawidłowa dana");
                }
                catch(ArrayIndexOutOfBoundsException ex) {
                    System.out.println(args[i] + " - liczba spoza zakresu");
                }
            }
        }
        catch(NumberFormatException ex) {
            System.out.println("nieprawidłowy zakres");
        }


    }
}

File: lista1/LiczbyPierwszetest.cpp
#include <iostream>
#include "LiczbyPierwsze.hpp"
using namespace std;

int main(int argc, char *argv[]) {
    try {
        int x = std::stoi(argv[1]);
        static LiczbyPierwsze* test = new LiczbyPierwsze(x);
        for (int i = 2; i<argc; i++){
            try {
                int n = std::stoi(argv[i]);
                if(test->liczba(x,n)!= -1){
                    cout<<n<<" - "<<test->liczba(x,n)<<endl;
                }
                else {
                    cout<<n<<" - liczba spoza zakresu"<<endl;
                }
            }
            catch (exception &err) {
                cout<<argv[i]<<" - nieprawidlowa dana"<<endl;
            }
        }
        delete test;
    }
    catch (exception &err) {
        cout<<"nieprawidlowy zakres"<<endl;
    }
}